osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -40,16 +40,30 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
 
             double sliderFactor = aimRating > 0 ? aimRatingNoSliders / aimRating : 1;
 
-            if (mods.Any(h => h is OsuModRelax))
-                speedRating = 0.0;
-
             double baseAimPerformance = Math.Pow(5 * Math.Max(1, aimRating / 0.0675) - 4, 3) / 100000;
             double baseSpeedPerformance = Math.Pow(5 * Math.Max(1, speedRating / 0.0675) - 4, 3) / 100000;
             double baseFlashlightPerformance = 0.0;
 
             if (mods.Any(h => h is OsuModFlashlight))
                 baseFlashlightPerformance = Math.Pow(flashlightRating, 2.0) * 25.0;
 
+            if (mods.Any(h => h is OsuModRelax))
+            {
+                // Similar idea to the PP system speed crosscheck, but for star rating.
+                double speedCrosscheck = baseAimPerformance / baseSpeedPerformance;
+                if (speedCrosscheck < 1.0) // From testing, higher values tend to be entirely unrelated or ridiculously difficult.
+                {
+                    double crosscheckMultiplier = Math.Min(1.0, 0.7 * speedCrosscheck);
+                    baseAimPerformance *= Math.Max(0.1, crosscheckMultiplier);
+                    baseSpeedPerformance *= Math.Max(0.1, crosscheckMultiplier);
+                }
+
+                // These are probably less than ideal, but can be worked on in the future.
+                baseAimPerformance *= 0.4;
+                baseSpeedPerformance *= 0.5;
+                baseFlashlightPerformance *= 0.6;
```

Comment:
Why are you scaling FL rating here? Pretty sure relax has nothing to do with FL

Reply:
In my opinion, tapping has atleast **some** factor to flashlight and the entire tapping portion is removed hence why it's scaled. If people disagree, I'm open to remove it.

Reply:
@MBmasher I think we need your opinion here